src/main/java/commandHandler/Add.java

```diff
@@ -0,0 +1,101 @@
+
+public class Add {
+    private static final String COMMAND_EVENT = "/event";
+    private static final String COMMAND_TODO = "/todo";
+    private static final String COMMAND_DEADLINE = "/deadline";
+    private static final String COMMAND_BY = "/by";
+    private static final String COMMAND_START = "/start";
+    private static final String COMMAND_END = "/end";
+
+    public static void addTask(String userInput) throws MissingCommandException {
+        String[] userInputArray = userInput.split(" ");
+        String command = userInputArray[0];
+        Task newTask;
+        /** Check if task description is left empty **/
+        if (userInputArray.length == 1) {
+            throw new MissingCommandException("Please enter a description for your task!");
+        }
+        /** Handle different task types **/
+        if (command.equals(COMMAND_TODO)) {
+            newTask = new Todo(userInput);
+        } else if (command.equals(COMMAND_DEADLINE)) {
+            if (!userInput.contains(COMMAND_BY)) {
+                throw new MissingCommandException("Please specify a deadline via the /by command!");
+            }
+            newTask = new Deadline(userInput);
+        } else if (command.equals(COMMAND_EVENT)) {
+            if (!userInput.contains(COMMAND_START) || !userInput.contains(COMMAND_END)) {
+                throw new MissingCommandException(
+                        "Please specify both start and end dates/times via the /start and /end commands!");
+            }
+            newTask = new Event(userInput);
+        } else {
+            return;
+        }
+        tasksList.addTask(newTask);
+        addTaskToStorage(newTask, tasksList.userTaskCount);
+        tasksList.userTaskCount++;
+    }
+
+    public static void addTaskToStorage(Task newTask, int taskCount) {
+        try {
+            FileWriter fw = new FileWriter(ProcessStorageTasks.FILE_PATH, true);
+            fw.write(Integer.toString(taskCount) + ":" + newTask.formattedString() + "\n");
+            fw.close();
+        } catch (IOException e) {
+            Display.warnUser(e.getMessage());
+        }
+    }
+
+    /*
+     * arr[0] -> index
+     * arr[1] -> task type
+     * arr[2] -> isDone
+     * arr[3] -> task description
+     * 
+     * For event:
+     * arr[4] -> start
+     * arr[5] -> end
+     * 
+     * For deadline:
+     * arr[4] -> cutoff
+     */
+
+    public static void addSavedTask(String arguments) {
+        Task newTask;
+        String[] taskStringArray = arguments.split(":");
+        String command = taskStringArray[1];
+        if (command.equals("Todo")) {
+            String inputFormat = "/todo " + taskStringArray[3];
+            newTask = new Todo(inputFormat);
+        } else if (command.equals("Deadline")) {
+            String inputFormat = "/deadline " + taskStringArray[3] + " /by " +
+                    taskStringArray[4];
+            newTask = new Deadline(inputFormat);
+        } else {
```

Comment:
Do not use else to execute the last option. Use else only for the intended default action.